libdnf/base/transaction.cpp

```diff
@@ -405,6 +406,9 @@ Transaction::TransactionRunResult Transaction::Impl::run(
     // finish history db transaction
     time = std::chrono::system_clock::now().time_since_epoch();
     db_transaction->set_dt_end(std::chrono::duration_cast<std::chrono::seconds>(time).count());
+    // TODO(jrohel): Also save the rpm db cookie to system state.
+    //               Possibility to detect rpm database change without the need for a history database.
```

Comment:
Where would you add the detection and how would you handle a mismatch though?

Reply:
According to the discussion with Kyle, it would be nice to report changes performed outside of DNF like YUM did including the diff.